src/server/game/Server/WorldSession.h

```diff
@@ -29,6 +29,7 @@
 #include "Packet.h"
 #include "QueryCallbackProcessor.h"
 #include "SharedDefines.h"
+#include "TradeData.h"
```

Comment:
Here is why I prefer to NOT use std::array, you aren't bloating includes

Reply:
Ah yes i mentioned this in the commit message as the following "I know people will not like the extra includes here, but to be safe we either need to move the enum or use another container to be safe."
We could always use a vector in this case and reserve the size, the C style stuff is really unsafe if someone comes around to touching it.

Reply:
Nobody is ever going to touch trade data size because its not a variable you can tweak, it must match client expectations (this can never change in 3.3.5 and in master branch I will be the person to do it, if it ever does (unlikely))

Reply:
Good points. I just want to keep the codebase up2date now that I'm doing TC for a hobby, which is where TC shines in comparision with the "competitors" (Way more up2date) :)

If trade size is a value that'll never change we could move it to shareddefines which is pretty much included everywhere already, and get rid of those new includes.

Reply:
Should we go with the "move the enum to shareddefines" then, include 1 less header and merge the PR ?

Reply:
Sounds like a good idea to me.

Reply:
No, I would leave the code as-is (until std::span)